src/charm.py

```diff
@@ -118,13 +118,13 @@ def _on_install(self, event: InstallEvent) -> None:
             return
         self._write_ue_config_file()
 
-    def _configure_sdcore_smf(self, event: EventBase) -> None:
+    def _configure_sdcore_smf(self, event: EventBase) -> None:  # noqa C901
```

Comment:
I do not think we should `noqa C901`. This means that the complexity of the method is too high because of too many branches.

I think we should find a way to split this instead. A quick idea would be to keep only the guard clauses in here and move line 160-164 to another method that gets called.

If that is not enough, we could refactor the pre-checks so that we do not get a bunch of conditions here. Something like:

```
@dataclass
class PrecheckResult:
    passed: bool
    defer: bool
    status: StatusBase
```

```
check_results = (result for results in (check1(), check2(), check3())) 
if not all(result.passed for result in check_results):
    # find the first failed result, set the status, defer if required and return
```